### PR TITLE
Migrating some tests to Async/Await

### LINKED ISSUE #569: Migrate tests to async/await syntax
As this is a relatively big change, the plan is to do it incrementally.

The following test could be used for some inspiration: https://github.com/hazelcast/hazelcast-nodejs-client/blob/9c1c2d10b8d5a2dc74fa21ddca3676a3e66cfb25/test/ClientRedoEnabledTest.js

### PR DESCRIPTION
**Pull request by sudiptog81**
Migrated the following tests to use the async/await syntax instead of Promises:

- `test/statistics/StatisticsTest.js`
- `test/rest_value/RestValueTest.js`
- `test/map/MapStoreTest.js`
- `test/MembershipListenerTest.js`
- `test/LoggingTest.js`
- `test/LifecycleServiceTest.js`
- `test/map/MapEntryProcessorTest.js`
- `test/map/MapPredicateTest.js`

Listener-related tests using `done()` have been left as-it-is. The test results were observed to be same, before and after changing each test. Some files modified in the commits in this PR were reverted to commit hash `bcf3a399ecb9d931aa1662802d634fadce5e8399` where this branch was made.

Related to #569.

### CODE REVIEW

File: test/statistics/StatisticsTest.js [hunk 1]
```diff
@@ -174,53 +157,36 @@ describe('StatisticsTest (non-default period)', function () {
...
-        }).then(function (st) {
-            stats1 = st;
-            return getClientStatisticsFromServer(cluster, client)
-        }).then(function (stats2) {
-            expect(stats1).to.be.equal(stats2);
-        });
+    it('should not change before period', async function () {
+        await TestUtil.promiseWaitMilliseconds(1000);
+        let stats1 = await getClientStatisticsFromServer(cluster, client);
+        let stats2 = await getClientStatisticsFromServer(cluster, client);
```
Comment (puzpuzpuz) [inline test/statistics/StatisticsTest.js:180]:
```suggestion
        const stats2 = await getClientStatisticsFromServer(cluster, client);
```

File: test/statistics/StatisticsTest.js [hunk 2]
```diff
@@ -174,53 +157,36 @@ describe('StatisticsTest (non-default period)', function () {
...
-            return TestUtil.promiseWaitMilliseconds(2000)
-        }).then(function () {
-            return getClientStatisticsFromServer(cluster, client);
-        }).then(function (stats2) {
-            expect(stats1).not.to.be.equal(stats2);
-        });
+    it('should change after period', async function () {
+        await TestUtil.promiseWaitMilliseconds(1000);
+        let stats1 = await getClientStatisticsFromServer(cluster, client);
+        await TestUtil.promiseWaitMilliseconds(2000)
```
Comment (puzpuzpuz) [inline test/statistics/StatisticsTest.js:187]:
```suggestion
        await TestUtil.promiseWaitMilliseconds(2000);
```

File: test/statistics/StatisticsTest.js [hunk 3]
```diff
@@ -174,53 +157,36 @@ describe('StatisticsTest (non-default period)', function () {
...
-            stats1 = st;
-            return TestUtil.promiseWaitMilliseconds(2000)
-        }).then(function () {
-            return getClientStatisticsFromServer(cluster, client);
-        }).then(function (stats2) {
-            expect(stats1).not.to.be.equal(stats2);
-        });
+    it('should change after period', async function () {
+        await TestUtil.promiseWaitMilliseconds(1000);
+        let stats1 = await getClientStatisticsFromServer(cluster, client);
```
Comment (puzpuzpuz) [inline test/statistics/StatisticsTest.js:186]:
```suggestion
        const stats1 = await getClientStatisticsFromServer(cluster, client);
```

File: test/statistics/StatisticsTest.js [hunk 4]
```diff
@@ -174,53 +157,36 @@ describe('StatisticsTest (non-default period)', function () {
...
+        await TestUtil.promiseWaitMilliseconds(2000)
+        let stats2 = await getClientStatisticsFromServer(cluster, client);
```
Comment (puzpuzpuz) [inline test/statistics/StatisticsTest.js:188]:
```suggestion
        const stats2 = await getClientStatisticsFromServer(cluster, client);
```

File: test/statistics/StatisticsTest.js [hunk 5]
```diff
@@ -229,34 +195,26 @@ describe('StatisticsTest (negative period)', function () {
...
 
-    it('should be enabled via configuration', function () {
-        return TestUtil.promiseWaitMilliseconds(1000).then(function () {
-            return getClientStatisticsFromServer(cluster, client);
-        }).then(function (stats) {
-            expect(stats).to.not.equal('');
-        });
+    it('should be enabled via configuration', async function () {
+        await TestUtil.promiseWaitMilliseconds(1000);
+        let stats = await getClientStatisticsFromServer(cluster, client);
```
Comment (puzpuzpuz) [inline test/statistics/StatisticsTest.js:217]:
```suggestion
        const stats = await getClientStatisticsFromServer(cluster, client);
```

**Comment by puzpuzpuz** [comment]
@sudiptog81 the changes look like the right direction overall. Could you address comments? Once you do it, I'll start the CI build.

**Comment by sudiptog81** [comment]
Sure, will definitely address all the concerns in the next commit. Thanks for your time!

File: .eslintrc.js [hunk 6]
```diff
@@ -19,7 +19,13 @@ module.exports = {
     parser: '@typescript-eslint/parser',
     plugins: [
         '@typescript-eslint',
+        'mocha'
```
[the remark this answers is not in the document]
Comment (sudiptog81) [reply .eslintrc.js:22]: Reverted changes to `eslint` and `package.json`.

**Review by puzpuzpuz — APPROVED** [review]
LGTM. Many thanks for the contribution!

**Comment by puzpuzpuz** [comment]
verify

**Pull request merged**